### PR TITLE
Feature/dev 6151

### PR DESCRIPTION
**Pull request by Atash3000**
## Testing Steps
This feature works on Maps and Charts (Bar Chart only).
To test on Charts:
1.  open Bar chart.
2. Add couple series
3. On Legend Panel in dropdown "Legend Style"  "gradient or boxes"
4. Set Position of Legend "TOP" or "BOTTOM" for gradient style
5. Select "Gradient Style" 'smooth" or "linear blocks.
6. Update add value to "Tick Rotations" field to change angle of ticks of the legend.
7. Perform same test on Maps

## Self Review
Screenshot 2024-08-18 at 15 00 01
Screenshot 2024-08-18 at 15 45 35
Screenshot 2024-08-18 at 19 57 53

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

### CODE REVIEW

File: packages/chart/src/components/EditorPanel/EditorPanel.tsx [hunk 1]
```diff
@@ -776,9 +777,27 @@ const EditorPanel = () => {
     return Object.keys(columns)
   }
 
-  const getDataValueOptions = data => {
+  const getLegendStyloptions = option => {
```
Comment (adamdoe) [inline packages/chart/src/components/EditorPanel/EditorPanel.tsx:780]: Review spelling and casing getLegendStyleOptions

File: packages/chart/src/components/Legend/Legend.Component.tsx [hunk 2]
```diff
@@ -22,29 +24,30 @@ export interface LegendProps {
   ref: React.Ref<() => void>
   seriesHighlight: string[]
   skipId: string
+  dimensions: [string, string]
```
Comment (adamdoe) [inline packages/chart/src/components/Legend/Legend.Component.tsx:27]: Add comment above describing what dimensions are for.

File: packages/chart/src/components/Legend/Legend.Component.tsx [hunk 3]
```diff
@@ -72,6 +75,12 @@ const Legend: React.FC<LegendProps> = forwardRef(({ config, colorScale, seriesHi
                   if (seriesHighlight.length > 0 && false === seriesHighlight.includes(itemName)) {
                     className.push('inactive')
                   }
+                  if (config.legend.style === 'gradient') {
```
Comment (adamdoe) [inline packages/chart/src/components/Legend/Legend.Component.tsx:78]: update TS on legend.

Comment (adamdoe) [reply packages/chart/src/components/Legend/Legend.Component.tsx:78]: Should you just return null at this point without updating the class? It looks like the next like is a return statement?

File: packages/map/src/CdcMap.tsx [hunk 4]
```diff
@@ -73,6 +74,9 @@ const countryKeys = Object.keys(supportedCountries)
 const countyKeys = Object.keys(supportedCounties)
 const cityKeys = Object.keys(supportedCities)
 
+// types
+import { type ViewportSize } from './types/MapConfig'
```
Comment (adamdoe) [inline packages/map/src/CdcMap.tsx:78]: Move up in the file.

File: packages/map/src/components/Modal.jsx
Comment (adamdoe) [inline packages/map/src/components/Modal.jsx:1]: Can you update this file to tsx

File: packages/map/src/components/DataTable.jsx
Comment (adamdoe) [inline packages/map/src/components/DataTable.jsx:1]: Update file to tsx

File: packages/core/helpers/useDataVizClasses.js
Comment (adamdoe) [inline packages/core/helpers/useDataVizClasses.js:1]: Update to ts

**Review by adamdoe — CHANGES_REQUESTED** [review]
Looking good! Can you make a few of the requested changes?

File: packages/chart/src/components/EditorPanel/EditorPanel.tsx (hunk 1 above)
Comment (Atash3000) [reply packages/chart/src/components/EditorPanel/EditorPanel.tsx:780]: Fixed

File: packages/chart/src/components/Legend/Legend.Component.tsx (hunk 3 above)
Comment (Atash3000) [reply packages/chart/src/components/Legend/Legend.Component.tsx:78]: Fixed

File: packages/map/src/CdcMap.tsx (hunk 4 above)
Comment (Atash3000) [reply packages/map/src/CdcMap.tsx:78]: Fixed

File: packages/chart/src/components/Legend/Legend.Component.tsx (hunk 2 above)
Comment (Atash3000) [reply packages/chart/src/components/Legend/Legend.Component.tsx:27]: Fixed

File: packages/map/src/components/DataTable.jsx
Comment (Atash3000) [reply packages/map/src/components/DataTable.jsx:1]: Fixed

File: packages/core/helpers/useDataVizClasses.js
Comment (Atash3000) [reply packages/core/helpers/useDataVizClasses.js:1]: Fixed

File: packages/map/src/components/Legend/components/Legend.tsx [hunk 5]
```diff
@@ -3,26 +3,31 @@ import { forwardRef, useContext } from 'react'
...
   skipId: string
+  currentViewport: ViewportSize
+  dimensions: any
 }
 
-const Legend = forwardRef((props, ref) => {
-  const { skipId } = props
+const Legend = forwardRef<HTMLDivElement, LegendProps>((props, ref) => {
+  const { skipId, currentViewport, dimensions } = props
+  let [mapWidth] = dimensions
```
Comment (joshlacey) [inline packages/map/src/components/Legend/components/Legend.tsx:27]: @Atash3000 this is causing an issue

**Pull request merged**